tests/libvmi/lifecycle.go

```diff
@@ -0,0 +1,117 @@
+package libvmi
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+)
+
+type VMIPool map[string]*v1.VirtualMachineInstance
+
+func NewVMIPool() VMIPool {
+	return map[string]*v1.VirtualMachineInstance{}
+}
+
+func (vmis VMIPool) Setup(loggedInExpecterFactory tests.VMIExpecterFactory) {
+	virtClient, err := kubecli.GetKubevirtClient()
+	ExpectWithOffset(1, err).NotTo(HaveOccurred(), "Should initialize KubeVirt client")
+
+	vmis.create(virtClient)
+	vmis.waitUntilReady(loggedInExpecterFactory, virtClient)
+}
+
+func (vmis VMIPool) create(virtClient kubecli.KubevirtClient) {
+	for name, vmi := range vmis {
+		vmi, err := virtClient.VirtualMachineInstance(vmi.GetNamespace()).Create(vmi)
```

Comment:
why is it preferred over vmi.Namespace ?
(its the same right ? just using meta.go IIUC)
usually we used vmi.Namespace for those

Reply:
I had some issues in the past when I tried to access the Name directly, skipping the method.

Overall, if there is a getter, it is probably for a reason. I would not skip directly to `.Namespace`.